WIP Add functional front-end testing

IshaVDave opened the pull request:
Purpose:
Testing various features of the web app layout-web-tool using Selenium Webdriver and Flask-testing.

IshaVDave commented:
A couple things:

1. There are some .vs files that got included into the PR and I'm not sure if you need them (they are meant only for some Visual Studio functionality right?). Once that is confirmed, I will remove them.
2. The code still doesn't work, and I'm in the process of getting it to.

SantiagoTorres commented on .vs/ProjectSettings.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "CurrentProjectSetting": null
+}
```
I think all of the .vs files shouldn't be checked in, as they are configuration settings from your editor. You probably want to add
```
.vs/*
```
to our gitignore so we don't add it in further commits. For now you want to remove these files from the commit and update the pr...

SantiagoTorres commented on tests/selenium_testing.py:
```diff
@@ -0,0 +1,36 @@
+    self.driver.quit()
+
+if __name__ == '__main__':
+    unittest.main()
```
In order for this to work, you need to add these requirements to `requirements.txt`

```
diff --git a/requirements.txt b/requirements.txt
index f23a780..26d84e5 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -2,3 +2,5 @@ Flask==1.1.2
 Flask-PyMongo==2.3.0
 Flask-WTF==0.14.3
 in-toto==0.5.0
+Flask-Testing
+selenium
```

SantiagoTorres reviewed (changes requested):
Pretty cool! 

![selenium](https://user-images.githubusercontent.com/3579933/94286255-c0fd0600-ff22-11ea-8bf9-897ff9a6cacd.gif)
I think we need a couple of minor tweaks before merging...

IshaVDave commented:
Thanks for the review! I have made the changes and everything should work now. Two more things:
1. Should I be rebasing the *selenium-testing file* and *updating gitignore and requirements.txt* into two commits for better history?
2. Should I keep adding tests or should it be a separate PR?

lukpueh commented:
> Thanks for the review! I have made the changes and everything should work now. Two more things:
> 
> 1. Should I be rebasing the _selenium-testing file_ and _updating gitignore and requirements.txt_ into two commits for better history?

IMO you can squash all 3 commits into one and add a nice description of your addition.

> 2. Should I keep adding tests or should it be a separate PR?

I'm fine either way. IIUC the PR doesn't test anything yet, right? Maybe you can add one initial basic test case to demonstrate how selenium works.

lukpueh commented on tests/selenium_testing.py:
```diff
@@ -0,0 +1,36 @@
+"""
+  Setting up Functional testing for layout web tool.
```
```suggestion
  Setting up functional testing for layout web tool.
```

lukpueh commented on tests/selenium_testing.py:
```diff
@@ -0,0 +1,36 @@
+  def setUp(self):
+    self.driver = webdriver.Chrome()
+
+  def test_Start(self):
```
```suggestion
  def test_start(self):
```

lukpueh reviewed (commented):
Very cool stuff, @IshaDave! Apart from two minor nits inline this looks good to go. 

It would also be nice to have a bit more code documentation, or an exemplary test case to make it easier for someone who wants to add tests. What do you think?

IshaVDave commented:
> IMO you can squash all 3 commits into one and add a nice description of your addition.

Will do!
> IIUC the PR doesn't test anything yet, right?

So a primary test that @SantiagoTorres and I had discussed was to see if the website opened and return a screenshot of the page, then check if you can proceed to the next and come back to this page, then return another screenshot. If the two screenshots are the same, the test runs successfully. I might have misunderstood that part of the instruction, but I am not able to find a way out to compare those two screenshots.

adityasaky commented:
@SantiagoTorres indicated this PR is ready for to be merged, @IshaDave. Could I ask you to squash some commits before we proceed? :smile:

IshaVDave commented:
> @SantiagoTorres indicated this PR is ready for to be merged, @IshaDave. Could I ask you to squash some commits before we proceed? 😄

Sounds great and will do!

IshaVDave committed:
Merge pull request #1 from in-toto/develop

merge

IshaVDave committed:
Add functional testing using Selenium Webdriver

Setting up the skeleton and initial tests for unit testing of front-end of layout-web-tool

IshaVDave committed:
Create a Testing Suite

Initialize suite of tests for layout-web-tool.

IshaVDave committed:
Update gitignore, travis and requirements

.gitignore :  ignore .vs files
requirements.txt : add Flask-testing and selenium
.travis.yml : add support for chromedriver installation and testing suite

adityasaky commented:
Thanks for the squashes, @IshaDave!
